src/uct/base/uct_mem.c

```diff
@@ -120,26 +120,33 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
                 alloc_length = length;
                 address      = uct_mem_alloc_params_get_address(params);
                 md           = params->mds.mds[md_index];
-                status = uct_md_query(md, &md_attr);
+                status = uct_md_query_v2(md, &md_attr);
                 if (status != UCS_OK) {
                     ucs_error("Failed to query MD");
                     goto out;
                 }
 
                 /* Check if MD supports allocation */
-                if (!(md_attr.cap.flags & UCT_MD_FLAG_ALLOC)) {
+                if (!(md_attr.flags & UCT_MD_FLAG_ALLOC)) {
                     continue;
                 }
 
                 /* Check if MD supports allocation with fixed address
                  * if it's requested */
                 if ((flags & UCT_MD_MEM_FLAG_FIXED) &&
-                    !(md_attr.cap.flags & UCT_MD_FLAG_FIXED)) {
+                    !(md_attr.flags & UCT_MD_FLAG_FIXED)) {
+                    continue;
+                }
+
+                /* Check if MD supports allocation of memory is exported if
+                 * it's requested */
```

Comment:
imo, this comment is less clear than the code below.  I'd remove it or did you mean
`Check if MD supports allocation of memory as exported if it's requested */`

Reply:
yes, I meant using `as` instead of `is`
fixed